providers/flagd/src/test/java/dev/openfeature/contrib/providers/flagd/resolver/process/targeting/FractionalTest.java

```diff
@@ -189,7 +189,49 @@ void invalidRuleSumNot100() throws JsonLogicEvaluationException {
         Object evaluate = fractional.evaluate(rule, data);
 
         // then
-        assertNull(evaluate);
+        assertEquals("blue", evaluate);
+    }
+
+    @Test
+    void invalidRuleSumlower100() throws JsonLogicEvaluationException {
+        // given
+        Fractional fractional = new Fractional();
+
+        /* Rule
+         *     [
+         *       [
+         *         "blue",
+         *         50
+         *       ],
+         *       [
+         *         "green",
+         *         30
+         *       ]
+         *     ]
+         * */
+
+        final List<Object> rule = new ArrayList<>();
+
+        final List<Object> bucket1 = new ArrayList<>();
+        bucket1.add("blue");
+        bucket1.add(50);
+
+        final List<Object> bucket2 = new ArrayList<>();
+        bucket2.add("green");
+        bucket2.add(30);
+
+        rule.add(bucket1);
+        rule.add(bucket2);
+
+        Map<String, String> data = new HashMap<>();
+        data.put(FLAG_KEY, "headerColor");
+        data.put(TARGET_KEY, "foo@foo.com");
```

Comment:
Maybe we should have a comment saying that the `target key` maps to the first variant

Reply:
I'm not sure what you mean exactly by this. Most likely, this is due to my lack of knowledge; I only copied and pasted the code, adapted input and expectations, and left the rest as is. Can you please elaborate?

Reply:
sure :) The [targeting key](https://openfeature.dev/docs/reference/concepts/evaluation-context/#targeting-key) is hashed and its value is used to put the user in always the same bucket when we do fractional evaluations. 
Therefore, the value `foo@foo.com` makes the fractional evaluation always return the first variant. Having this as a comment will make the testing easier to follow and understand why we always expect the first variant to be returned :)